script.js

```diff
@@ -30,6 +32,39 @@ function setStyles(element) {
 	element.style.left = randInRange(0,leftMax) + 'px';
 	element.style.top = randInRange(0,topMax) + 'px';
 	element.style.backgroundColor = 'rgba(' + randInRange(0,256) + ',' + randInRange(0,256) + ',' + randInRange(0,256) + ',' + Math.random() + ')';
+
+	// using setTimeout, run setStyles on the box once every second, continuously
+	/*var repeater = setTimeout(function(){
+		setStyles(element);
+	},1000);*/
+}
+
+// create a new loop that smoothly animates the position of the box, moving it 60 times per second
+function moveBox(element) {
+    var leftPos = 0,
+    	topPos = 0,
+		divWidth = element.offsetWidth,
+		divHeight = element.offsetHeight,
+		leftMax = windowWidth - divWidth,
+		topMax = windowHeight - divHeight,
+		velocity = randInRange(0,15); // randomize velocity
+
+    leftPos += (element.offsetLeft - element.scrollLeft + element.clientLeft);
+    topPos += (element.offsetTop - element.scrollTop + element.clientTop);
+
+    var moveRepeat = setTimeout(function(){
+    	moveBox(element);
+    }, velocity);
+
+    if (leftPos == leftMax || topPos == topMax) {
+        clearTimeout(moveRepeat);
+    } else {
+        leftPos++; 
+        topPos++; 
+        element.style.left = leftPos + 'px'; 
+        element.style.top = topPos + 'px';
+    }
```

Comment:
More efficient to call the timeout here in the success condition rather than call it and then abort it in the fail condition. In the range of 0-15milliseconds, it's possible we'd miss the opportunity to cancel the timeout.